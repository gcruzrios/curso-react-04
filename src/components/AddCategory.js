import React, { useState } from 'react'
import PropTypes from 'prop-types';



export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('')

    const handleInputChange = (e) => {
        //console.log(e.target.value)
        setInputValue(e.target.value);

        //console.log ('handleInputChange llamado');
    }

    const handleSubmit = (e)=> {
        e.preventDefault();
        //console.log('handleSubmit', inputValue);
        if ( inputValue.trim().length > 2 ){
            setCategories( categories =>[inputValue, ...categories]);
            setInputValue('');
        }
        
    }

    return (
        <>
            {/* <h2>Addd Category</h2> */}
             {/* <h1>{ inputValue }</h1> */}
             <form onSubmit={ handleSubmit }>
                 <p>{ inputValue }</p>
                <input 
                    type="text"
                    value={ inputValue} 
                    onChange={handleInputChange }
                />  
                
             </form>
             
        </>
    )
}

AddCategory.propTypes={
    setCategories: PropTypes.func.isRequired
}
