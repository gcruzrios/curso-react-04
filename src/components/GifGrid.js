import React from 'react'
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';


export const GifGrid = ({category}) => {

    //const [count, setCount] = useState(0)

    
    const { data:images, loading } = useFetchGifs( category );
    //console.log(state)
    
    
    /* 

        useEffect( ()=> {
        getGifs( category ).then (imgs => setImages( imgs ));
        }, [ category ]) */


    
    

    return (
        <>
            <h3 className='animate__animated animate__fadeIn'>{category}</h3> 
            
            { loading && <p className='animate__animated animate__flash'> Loading </p> }
            
             <div className='card-grid animate__animated animate__fadeIn'>
                
                    {
                        images.map ( img => (
                            <GifGridItem 
                            key = { img.id }
                            {...img}
                            //img = { img } 
                            />
                        ))
                    }
                
            </div> 
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}