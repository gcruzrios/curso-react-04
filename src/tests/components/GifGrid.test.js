import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs')

describe('pruebas en el < GifGrid />', () => {
    
    const category = 'One Punch';
    
    

    test('Debe de mostrarse correctamente', () => {

        useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        });

        const wrapper = shallow ( <GifGrid category={ category } /> );
        expect (wrapper).toMatchSnapshot();
    });
    
    test('debe mostrar items cuando se cargan imagenes useFetchGifs ', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://google.com',
            title: 'ABC Title'
        },
        {
            id: '123',
            url: 'https://google.com',
            title: 'ABC Title'
        }
    
        ];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        // 
        const wrapper = shallow ( <GifGrid category={ category } /> );
        expect(wrapper).toMatchSnapshot();
        expect (wrapper.find('p').exists()).toBe(false);
        expect (wrapper.find('GifGridItem').length).toBe(gifs.length);

    })
    
    
})
