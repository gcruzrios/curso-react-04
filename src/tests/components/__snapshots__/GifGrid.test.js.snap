// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el < GifGrid /> Debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
     Loading 
  </p>
  <div
    className="card-grid animate__animated animate__fadeIn"
  />
</Fragment>
`;

exports[`pruebas en el < GifGrid /> debe mostrar items cuando se cargan imagenes useFetchGifs  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid animate__animated animate__fadeIn"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="ABC Title"
      url="https://google.com"
    />
    <GifGridItem
      id="123"
      key="123"
      title="ABC Title"
      url="https://google.com"
    />
  </div>
</Fragment>
`;
